compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/interop/JMethodResolver.java

```diff
@@ -238,19 +240,17 @@ private boolean hasEquivalentPathParamCount(JMethodRequest jMethodRequest, JMeth
     }
 
     private boolean hasEquivalentFunctionParamCount(JMethodRequest jMethodRequest, JMethod jMethod) {
-        // Currently, this is only applicable for resource and remote methods which have empty path parameters.
-        // If path parameters are present, this will be handled by 'hasEquivalentPathAndFunctionParamCount' method
-        // by bundling both.
-        if (jMethodRequest.receiverType == null || jMethodRequest.pathParamCount != 0
-                || jMethodRequest.bParamTypes.length == 0) {
-            return false;
-        }
+        // This is only applicable for resource and remote methods which have at least one function
+        // parameter other than path parameters and the bundling of path parameters is not required.
         Class<?>[] paramTypes = jMethod.getParamTypes();
-        boolean isFirstParamServiceObject = jMethodRequest.bParamTypes[0].tag == TypeTags.SERVICE;
-        // Get the count of parameters of the resource/remote methods by excluding the receiver type.
-        int count = isFirstParamServiceObject ? paramTypes.length - 1 : paramTypes.length;
-        int reducedParamCount = isFirstParamServiceObject ? 2 : 1;
-        if (count < reducedParamCount || count > reducedParamCount + 2) {
+        int count = paramTypes.length;
+        int reducedParamCount = jMethodRequest.pathParamCount + 1;
+        int functionParamCount = jMethodRequest.bFuncParamCount - jMethodRequest.pathParamCount;
+        // TODO: Remove 'Symbols.isFlagOn(jMethodRequest.bParamTypes[0].flags, Flags.SERVICE)' check after fixing
+        //  https://github.com/ballerina-platform/ballerina-lang/issues/42456.
+        if (jMethodRequest.receiverType == null || functionParamCount < 1
+                || count < reducedParamCount || count > reducedParamCount + 2
+                || Symbols.isFlagOn(jMethodRequest.bParamTypes[0].flags, Flags.SERVICE)) {
```

Comment:
Need to remove this check once addressed the issue https://github.com/ballerina-platform/ballerina-lang/issues/42456. The logic used to bundle the parameters clashes with following use cases without the introduced check. https://github.com/ballerina-platform/module-ballerina-graphql/blob/8bd564d7d2eb2513a377bc58cc5d94a7d762803d/ballerina/engine.bal#L427 Anyway, Specifying the parameter annotation will resolve the conflicts.